### PR TITLE
Improve docs

### PR DESCRIPTION
**Pull request by dgageot**
+ Improve the static pages
+ Improve the Go structs to make them more friendly to the documentation generator (test section and profile patches)
+ fixed comments here and there

### CODE REVIEW

File: docs/content/en/docs/how-tos/builders/_index.md [hunk 1]
```diff
@@ -9,66 +9,77 @@ to build Docker images.
...
-installed on your machine, you can configure Skaffold to build artifacts with
-the local Docker daemon. 
+installed, Skaffold can be configured to build artifacts with the local
+Docker daemon. 
 
 By default, Skaffold connects to the local Docker daemon using
-[Docker Engine APIs](https://docs.docker.com/develop/sdk/). You can, however,
-ask Skaffold to use the [command-line interface](https://docs.docker.com/engine/reference/commandline/cli/)
+[Docker Engine APIs](https://docs.docker.com/develop/sdk/). Skaffold can, however,
+be aked to use the [command-line interface](https://docs.docker.com/engine/reference/commandline/cli/)
```
Comment (balopat) [inline docs/content/en/docs/how-tos/builders/_index.md]:
```suggestion
be asked to use the [command-line interface](https://docs.docker.com/engine/reference/commandline/cli/)
```

File: docs/content/en/docs/how-tos/builders/_index.md [hunk 2]
```diff
@@ -9,66 +9,77 @@ to build Docker images.
...
+After the artifacts are successfully built, Docker images will be pushed
+to the remote registry. You can choose to skip this step.
 
-The `local` type offers the following options:
+### Configuration
+
+To use the local Docker daemon, add build type `local` to the `build` section
+of `skaffold.yaml`. The following options can optionally be configured:
 
 {{< schema root="LocalBuild" >}}
```
Comment (balopat) [inline docs/content/en/docs/how-tos/builders/_index.md]: looking at this rendered it is a bit weird that the default is empty for `push` - shouldn't it be `false` there too? 

[image](http://35.236.109.79:1313/docs/how-tos/builders/)

Comment (dgageot) [reply docs/content/en/docs/how-tos/builders/_index.md]: In fact it depends on the k8s context.

Comment (balopat) [reply docs/content/en/docs/how-tos/builders/_index.md]: oh yeah, this is due to a previous merge - I guess it's a more complex logic as currently on skaffold.dev it is manually specified as "false for local clusters, true for remote clusters."...

**Comment by dgageot** [comment]
Let me fix the typo

**Comment by dgageot** [comment]
Travis is green but can't seem to get the status updated. Let's merge this PR

**Pull request merged**